File: Leetcode 3016 Minimum Number of Pushes to Type Word II.py
## 2026/07/30

class Solution:
    def minimumPushes(self, word: str) -> int:
        dic = Counter(word)
        freq = sorted(dic.values(), reverse = True)
        
        count = 0
        for index, q in enumerate(freq):
            count += q * math.ceil((index+1)/8)

        return count

File: Leetcode 3016 Minimum Number of Pushes to Type Word II.go
// 2026/07/30
func minimumPushes(word string) int {
    freq := make(map[rune]int)

    for _, c := range word {
        freq[c] ++
    }

    counts := make([]int, 0, len(freq))
    for _, v := range freq{
        counts = append(counts, v)
    }

    slices.SortFunc(counts, func(a, b int) int { return b - a })
    ans := 0
    for index, value := range counts{
        ans += value * (index/8+1) 
    }

    return ans
}
